examples/profile-picture/src/app/_components/profile-pic-uploader.tsx

```diff
@@ -0,0 +1,315 @@
+        </CardHeader>
+
+        {!file && (
+          <div className="relative p-6">
+            {newImageLoading && (
+              <div className="absolute inset-6 flex animate-pulse items-center justify-center rounded-2xl bg-black/80" />
+            )}
+            <img
+              src={output?.preview ?? props.user.image ?? "/fallback.svg"}
+              onClick={() => inputRef.current?.click()}
+              alt="Profile Picture"
+              className={"size-32 cursor-pointer rounded-2xl hover:opacity-75"}
+            />
+          </div>
+        )}
+      </div>
+      {file && (
+        <CardContent>
+          <div className="relative h-full min-h-[400px] w-full">
+            <Button
+              size="icon"
+              variant="destructive"
+              className="z-100 absolute right-2 top-2"
+              onClick={() => {
+                setFile(null);
+                if (inputRef.current) inputRef.current.value = "";
+              }}
+            >
+              <XIcon className="size-5" />
+            </Button>
+            <Cropper
+              maxZoom={5}
+              image={file?.preview}
+              aspect={imageProperties?.aspectRatio}
+              crop={crop}
+              zoom={zoom}
+              objectFit="contain"
+              onCropChange={setCrop}
+              onZoomChange={setZoom}
+              onCropComplete={(_, area) => setCroppedArea(area)}
+            />
+          </div>
+          {/* {output && (
+            <div
+              className="mt-2 flex w-full items-center justify-center"
+              style={{ height: imageProperties?.height ?? 200 }}
+            >
+              <img
+                src={output.preview}
+                alt="Cropped image"
+                className="border"
+              />
+            </div>
+          )} */}
+        </CardContent>
+      )}
+      <CardFooter className="justify-between border-t px-6 py-4">
+        <p className="text-muted-foreground text-sm">
+          {file
+            ? "Zoom and Drag to crop the image."
+            : props.user.image
+              ? "Click on the profile picture to upload a new one."
+              : "You do not yet have a profile picture, click on the placeholder to upload one."}
+        </p>
+        <div className="flex items-center gap-2">
+          {file && !isUploading && (
+            <Button
+              size="sm"
+              variant="secondary"
+              onClick={() => inputRef.current?.click()}
+            >
+              <ImageIcon className="size-5" />
+            </Button>
+          )}
+          {file && (
+            <Button
+              size="sm"
+              onClick={uploadCroppedImage}
+              disabled={isUploading}
+            >
+              {isUploading && <Loader2 className="mr-2 h-4 w-4 animate-spin" />}
+              Save
+            </Button>
+          )}
+        </div>
+      </CardFooter>
+    </Card>
+  );
+}
+
```

Comment:
@markflorkowski From here on below are generic helper functions. Are they worthwhile exporting from uploadthing somewhere? Or should we keep it all one big example for now?

Reply:
I think leaving them as part of the example is good for now. Makes sense to export alongside the components once we ship them. 

`expandImageProperties` might make sense to export from `@uploadthing/shared`  now though. (feels similar to eg. `generatePermittedFileTypes`)